src/client/mesh.cpp

```diff
@@ -199,11 +200,12 @@ void setMeshColor(scene::IMesh *mesh, const video::SColor &color)
 
 void setMeshBufferTextureCoords(scene::IMeshBuffer *buf, const v2f *uv, u32 count)
 {
-	const u32 stride = getVertexPitchFromType(buf->getVertexType());
+	assert(buf->getVertexType() == video::EVT_STANDARD);
```

Comment:
This code only makes sense for `EVT_STANDARD` because we treat `vertices` as an array of `S3DVertex` below and we can't really make assumptions about struct layout here.

The only usage of this function is in the upright sprite code, where the type will always be `EVT_STANDARD`, so this is fine.

Reply:
maybe just copy it there then?